src/main/java/carparks/parking/FeeCalculator.java

```diff
@@ -0,0 +1,56 @@
+package carparks.parking;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+@Component
+class FeeCalculator {
+
+	public BigDecimal calculateFee(ParkingEntry entry) {
+
+		LocalDateTime start = entry.getStart();
+		LocalDateTime end = (entry.getFinish() == null) ? LocalDateTime.now() : entry.getFinish();
+		Duration duration = Duration.between(start, end);
+		
+		long durationInHours = duration.getSeconds() / 3600;
+		
+		// if the duration is even one second longer than the full hour
```

Comment:
A nie wystarczyłoby użyć `ceil`? Czemu te podejście jest lepsze?

Reply:
Oczywiście użycie ceil by wystarczyło. Nie wiedziałem, że taka funkcja istnieje - jestem trochę za bardzo przyzwyczajony do niskopoziomowego programowania.

Reply:
Ogólnie z `ceil` jest problem przy bardzo dużych liczbach i myślałem, że dlatego jej nie użyłeś (https://stackoverflow.com/questions/7139382/java-rounding-up-to-an-int-using-math-ceil), ale w tym przpadku by się sprawdziła.